src/engine/strat_engine/backstore/devices.rs

```diff
@@ -237,9 +239,32 @@ pub fn initialize_devices(
         dev_info: &DeviceInfo,
         pool_uuid: PoolUuid,
         mda_data_size: MDADataSize,
+        key_description: Option<&str>,
     ) -> StratisResult<StratBlockDev> {
-        let mut f = OpenOptions::new().write(true).open(&dev_info.devnode)?;
+        let dev_uuid = Uuid::new_v4();
+        let (path, deviceno) = match key_description {
+            Some(desc) => {
+                let path = initialize_encrypted_stratis_device(
```

Comment:
Does the call to ```initialize_encrypted_stratis_devices```, if it fails, leave anything that can or should be cleaned up? If not, can you add a comment indicating why not?

Reply:
I'll take a closer look at this as we may leave just the devicemapper device around and I could clean that up. The `.disown()` method should take care of that for all previously activated devices so I'll double check that that happens.

Reply:
This should be mostly resolved.